basket/Basket.js

```diff
@@ -285,6 +302,13 @@ export default class Basket {
     ) {
       if (showSpinner) EventBus.$emit('spinnerShow')
 
+      // check that selected date and timespan is within available dates
+      const checkResult = this.checkAvailableDates([basketEntryInstance])
+      if (!checkResult) {
+        EventBus.$emit('spinnerHide')
+        return false
+      }
```

Comment:
lots of duplicate code aswell, you have the same logic 3 times here now

Reply:
How to improve? If the check fails, the function has to exit and return a boolean..

Reply:
@michael-scheurer it cant be that the logic has to check for this behaviour 3 times in a row. If it indeed does, you should split up the function it is in into 3 separate functions, as it makes it obvious that the current function is handling 3 different usecases separately.